go/types/list.go

```diff
@@ -185,18 +219,22 @@ func (l List) IterAll(f listIterAllFunc) {
 	})
 }
 
+// Iterator returns a ListIterator which can be used to iterate efficiently over a list.
 func (l List) Iterator() ListIterator {
 	return l.IteratorAt(0)
 }
 
+// IteratorAt returns a ListIterator starting at index.
 func (l List) IteratorAt(index uint64) ListIterator {
 	return ListIterator{newCursorAtIndex(l.seq, index)}
 }
 
+// Diff returns the diff of two different lists.
 func (l List) Diff(last List, changes chan<- Splice, closeChan <-chan struct{}) {
 	l.DiffWithLimit(last, changes, closeChan, DEFAULT_MAX_SPLICE_MATRIX_SIZE)
 }
 
+// DiffWithLimit returns a partial diff of two different lists.
```

Comment:
The `limit` is how big of an edit distance matrix we are willing to compute versus just saying the thing changed.